### PR TITLE
Manage multiple file upload

### PR DESCRIPTION
**Pull request by cp-dharti-r**
> PR Description:
- Made a component with considering custom design(by user) for managing multiple file upload with preview
- Implement call-back functionality for showing the process/uploading view of files
- Refactor single file upload management component
- Exported single file upload and multiple file upload management components
- Update `README.md` file for considering whole library functionality step by step
- Tested library by linking npm in the local

> Test Steps:
- You can check the `README.md` on https://github.com/canopas/file-upload-web-frontend/tree/upload-multiple-files
- For testing the component, checkout this branch and run the server using `npm run dev`
- You can comment and uncomment the code on the `IndexView.vue` file for testing both modules

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,46 +1,373 @@
-# File management
+# File Management with Preview
 
-This template should help get you started developing with Vue 3 in Vite.
+A versatile and user-friendly file management system built with Vue.js and TypeScript that allows for single file uploading and multiple file management with a preview feature. It allows you to select files and preview them, returning an array of selected files. You can provide custom design for your file component and also, you can over ride it's class.
```
Comment (cp-sumi-k) [inline README.md]: Can be replaced with following

`A versatile and user-friendly file management system built with Vue.js and TypeScript that allows for single and multiple file uploading with a preview feature. It allows you to select files and preview them, returning an array of selected files. You can provide custom design and override classes for your file component.`

File: README.md [hunk 2]
```diff
@@ -1,46 +1,373 @@
...
-```sh
-npm run dev
+## Usage
+
+To manage and preview files with this library, follow these steps:
+
+### Import the library into your file
+
+```js
+const { SingleFileUpload, MultipleFileUpload } = require('file-upload-frontend')
```
Comment (cp-sumi-k) [inline README.md:60]: Can add like for `common.js`  and for `esModules`

File: README.md [hunk 3]
```diff
@@ -1,46 +1,373 @@
...
+
+OR
+
+import { SingleFileUpload, MultipleFileUpload } from 'file-upload-frontend'
+
+import { SingleFileUpload } from 'file-upload-frontend' // you can import only one component as per your requirement
```
Comment (cp-sumi-k) [inline README.md]: No need to mention this

File: README.md [hunk 4]
```diff
@@ -1,46 +1,373 @@
...
+    SingleFileUpload
+  },
+  data() {
+    return {
+      received: {} as Object,
+      isUploading: false // it's a required thing to pass in component
+    }
+  },
+  methods: {
+    receiveFile(file: any) {
```
Comment (cp-sumi-k) [inline README.md:86]: When this method will call?

File: README.md [hunk 5]
```diff
@@ -1,46 +1,373 @@
...
+  - :removeBtn="'remove'" - Provide text for remove button, it will show default `x` if you are not using this props.
+  - :uploadBtn="'Upload'" - Provide text for save or upload files button, it will show default `Upload` if you are not using this props.
+  - :progressBtn="'Uploading...'" - Provide text for button when files under the process, it will show default `Uploading...` if you are not using this props.
+  - :isUploading="isUploading" - This is the `required` props. It will be default `false` and when you got the array of files and successfully uploaded all the files,
+    you need to make it `false` again, as we are making it `true` on button click.
+  - :accept="'image/jpg, image/jpeg, image/png, video/mp4, audio/mp3, application/zip'" - You can add frontend validation for selectiong the required type of file. 
+    By default it will select all the type of files.
+  - Dynamic images according to file types. When you select below described file, it will show your given image. (It's not required, we will show default image)
+    - :pdfImg="'../assets/images/pdf-icon.png'" - For pdf files
+    - :textImg="'../assets/images/text-icon.png'" - For text files
```
Comment (cp-sumi-k) [inline README.md]: How can we pass multiple files using these properties? What if I have multiple text files?

File: src/App.vue [hunk 6]
```diff
@@ -1,8 +1,8 @@
 <script setup lang="ts">
 import { RouterView } from 'vue-router'
-import "@/assets/css/style.css"
+// import "../assets/css/style.css"
```
Comment (cp-sumi-k) [inline src/App.vue]: Let;s remove it if not required

File: src/components/MultipleFile.vue [hunk 7]
```diff
@@ -0,0 +1,254 @@
+<template>
+  <div class="d-flex container">
+    <!-- <div v-for="(img, k) in images" :key="k" class="d-flex mb-3">
```
Comment (cp-sumi-k) [inline src/components/MultipleFile.vue]: Let's remove unused code from everywhere

File: src/components/MultipleFile.vue [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+          type="button"
+          class="absolute text-white bg-danger border-danger rounded-circle remove-btn"
+          @click="removeImg(k)"
+        >
+          x
+        </button>
+      </div>
+    </div> -->
+
+    <div v-if="inputs.length > 0">
```
Comment (cp-sumi-k) [inline src/components/MultipleFile.vue]: Make sure the design is responsive and works with touch also

**Review by cp-sumi-k — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (cp-dharti-r) [reply README.md:86]: @file="receiveFile"
- when we click on upload button it will return file object in it

File: README.md (hunk 5 above)
Comment (cp-dharti-r) [reply README.md]: This is not for real file, as we can't show some content in preview. like for text file musiz file. so, it's for placeholder image where are showing in preview. like,
image

Comment (cp-sumi-k) [reply README.md]: Then i think we need to rename properties like pdfPreview or something else.
It is currently misleading

**Review by cp-sumi-k — APPROVED** [review]

**Pull request merged**